src/libsystemd/sd-netlink/netlink-util.h

```diff
@@ -135,7 +135,21 @@ static inline int netlink_message_append_hw_addr(sd_netlink_message *m, unsigned
         return sd_netlink_message_append_hw_addr(m, type, data->bytes, data->length);
 }
 
-int netlink_message_append_in_addr_union(sd_netlink_message *m, unsigned short type, int family, const union in_addr_union *data);
+static inline int netlink_message_append_in_addr_union(
+                sd_netlink_message *m,
+                unsigned short type,
+                int family,
+                const union in_addr_union *data) {
+        switch (family) {
+        case AF_INET:
+                return sd_netlink_message_append_in_addr(m, type, &data->in);
+        case AF_INET6:
+                return sd_netlink_message_append_in6_addr(m, type, &data->in6);
+        default:
```

Comment:
How about to simply call `netlink_message_append_addr(m, type, family, data->bytes)`?